server/rest/src/main/java/org/infinispan/rest/resources/TasksResource.java

```diff
@@ -0,0 +1,104 @@
+            .invocation().methods(GET).path("/v2/tasks/{taskName}").withAction("exec").handleWith(this::runTask)
+            .create();
+   }
+
+   private CompletionStage<RestResponse> listTasks(RestRequest request) {
+      EmbeddedCacheManager cacheManager = invocationHelper.getRestCacheManager().getInstance();
+      TaskManager taskManager = SecurityActions.getGlobalComponentRegistry(cacheManager).getComponent(TaskManager.class);
+      NettyRestResponse.Builder builder = new NettyRestResponse.Builder();
+      try {
+         List<Task> tasks = taskManager.getTasks();
+         byte[] resultBytes = invocationHelper.getMapper().writeValueAsBytes(tasks);
+         builder.contentType(APPLICATION_JSON_TYPE).entity(resultBytes);
+      } catch (JsonProcessingException e) {
+         builder.status(HttpResponseStatus.INTERNAL_SERVER_ERROR).entity(e.getMessage());
+      }
+      return completedFuture(builder.build());
+   }
+
+   private CompletionStage<RestResponse> createScriptTask(RestRequest request) {
+      String taskName = request.variables().get("taskName");
+      EmbeddedCacheManager cacheManager = invocationHelper.getRestCacheManager().getInstance();
+      ScriptingManager scriptingManager = SecurityActions.getGlobalComponentRegistry(cacheManager).getComponent(ScriptingManager.class);
+      NettyRestResponse.Builder builder = new NettyRestResponse.Builder();
+      ContentSource contents = request.contents();
+      byte[] bytes = contents.rawContent();
+      MediaType sourceType = request.contentType() == null ? APPLICATION_JAVASCRIPT : request.contentType();
+      String script = StandardConversions.convertTextToObject(bytes, sourceType);
+      scriptingManager.addScript(taskName, script);
+      return completedFuture(builder.build());
+   }
+
+   private CompletionStage<RestResponse> runTask(RestRequest request) {
+      String taskName = request.variables().get("taskName");
+      EmbeddedCacheManager cacheManager = invocationHelper.getRestCacheManager().getInstance();
+      TaskManager taskManager = SecurityActions.getGlobalComponentRegistry(cacheManager).getComponent(TaskManager.class);
+      TaskContext taskContext = new TaskContext();
+      request.parameters().forEach((k, v) -> {
+         if (k.startsWith("param.")) {
+            taskContext.addParameter(k.substring(6), v.get(0));
+         }
+      });
+      return taskManager.runTask(taskName, taskContext).thenApply(result -> {
+         NettyRestResponse.Builder builder = new NettyRestResponse.Builder();
+         try {
+            if (result instanceof byte[]) {
+               builder.contentType(TEXT_PLAIN_TYPE).entity(result);
+            } else {
+               byte[] resultBytes = invocationHelper.getMapper().writeValueAsBytes(result);
+               builder.contentType(APPLICATION_JSON_TYPE).entity(resultBytes);
+            }
+         } catch (JsonProcessingException e) {
+            builder.status(HttpResponseStatus.INTERNAL_SERVER_ERROR);
```

Comment:
don't you want to add the exception message to the response?

Reply:
Done, thanks